include/nvexec/nvtx.cuh

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2023 NVIDIA Corporation
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <nvtx3/nvToolsExt.h>
+
+#include "../stdexec/execution.hpp"
+#include <type_traits>
+
+#include "stream/common.cuh"
+
+namespace nvexec {
+
+  namespace STDEXEC_STREAM_DETAIL_NS { namespace nvtx {
+
+    enum class kind {
+      push,
+      pop
+    };
+
+    template <kind Kind, class ReceiverId>
+    struct receiver_t {
+      class __t : public stream_receiver_base {
+        using Receiver = stdexec::__t<ReceiverId>;
+        using Env = typename operation_state_base_t<ReceiverId>::env_t;
+
+        operation_state_base_t<ReceiverId>& op_state_;
+        std::string name_;
+
+       public:
+        using __id = receiver_t;
+
+        template <__one_of<set_value_t, set_error_t, set_stopped_t> Tag, class... As>
+        friend void tag_invoke(Tag tag, __t&& self, As&&... as) noexcept {
+          if constexpr (Kind == kind::push) {
+            nvtxRangePushA(self.name_.c_str());
+          } else {
+            nvtxRangePop();
+          }
+
+          self.op_state_.propagate_completion_signal(tag, (As&&) as...);
+        }
+
+        friend Env tag_invoke(get_env_t, const __t& self) noexcept {
+          return self.op_state_.make_env();
+        }
+
+        explicit __t(operation_state_base_t<ReceiverId>& op_state, std::string name)
+          : op_state_(op_state)
+          , name_(name) {
+        }
+      };
+    };
+
+    template <kind Kind, class SenderId>
+    struct nvtx_sender_t {
+      using Sender = stdexec::__t<SenderId>;
+
+      struct __t : stream_sender_base {
+        using __id = nvtx_sender_t;
+        Sender sndr_;
+        std::string name_;
+
+        using _set_error_t = completion_signatures<set_error_t(cudaError_t)>;
+
+        template <class Receiver>
+        using receiver_t = stdexec::__t<receiver_t<Kind, stdexec::__id<Receiver>>>;
+
+        template <class... Tys>
+        using _set_value_t = completion_signatures<set_value_t(Tys...)>;
+
+        template <class Self, class Env>
+        using _completion_signatures_t = //
+          __try_make_completion_signatures<
+            __copy_cvref_t<Self, Sender>,
+            Env,
+            _set_error_t,
+            __q<_set_value_t>>;
```

Comment:
Because of defaults, this is equivalent.

```suggestion
        template <class Self, class Env>
        using _completion_signatures_t = //
          __try_make_completion_signatures<
            __copy_cvref_t<Self, Sender>,
            Env,
            _set_error_t>;
```